src/uwtools/rocoto.py

```diff
@@ -182,13 +182,60 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        for key, block in config.items():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config={key: block})
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={key: block})
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, config={key: block})
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
```

Comment:
If you pass in `{tag: block}` here instead of `{key: block}`, it might simplify what you need to reprocess in the helper method (suggestion is untested):
```
def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
    for tag, block in config.items():
        self._set_attrs(SubElement(e, tag), block)
```

Edit: If you passed `key` and `block` as two separate arguments, it would simplify even further to:

```
def _add_task_dependency_operand(self, e: Element, key, block):
    self._set_attrs(SubElement(e, tag), block)
```